### PR TITLE
Adding inspec init profile for GCP.

### PR DESCRIPTION
**Pull request by skpaterson**
An attempt to address https://github.com/inspec/inspec-gcp/issues/61 and give GCP users a nice profile generator. 

Signed-off-by: Stuart Paterson <spaterson@chef.io>

### CODE REVIEW

**Review by ghost — APPROVED** [review]
I like this, will likely add something similar for Inspec-Azure if added!

**Review by miah — APPROVED** [review]
Thanks @skpaterson

File: lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/attributes.yml [hunk 1]
```diff
@@ -0,0 +1 @@
+# gcp_project_id: 'your-gcp-project'
```
Comment (jerryaldrichiii) [inline lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/attributes.yml]: Since this is commented, can we remove it?

**Review by jerryaldrichiii — CHANGES_REQUESTED** [review]

File: lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/README.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
+# Example InSpec Profile For GCP
+
+This example shows the implementation of an InSpec profile for GCP that depends on the [InSpec GCP Resource Pack](https://github.com/inspec/inspec-gcp).  See the [README](https://github.com/inspec/inspec-gcp) for instructions on setting up appropriate GCP credentials.
+
+##  Create a profile 
+
+```
+$ inspec init profile_gcp gcp-profile
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/README.md]: I don't like this CLI interface.  I think it should be:

`inspec init profile --platform gcp my-profile`

I think the `inspec-init` plugin should be modified to accommodate that.

File: lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/inspec.yml [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+attributes:
+- name: gcp_project_id
+  required: true
+  description: 'The GCP project identifier.'
+  type: string
+depends:
+- name: inspec-gcp
+  url: https://github.com/inspec/inspec-gcp/archive/master.tar.gz
+supports:
+- platform: gcp
```
Comment (clintoncwolfe) [inline lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/inspec.yml]: 💯 Excellent!

File: lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/README.md (hunk 2 above)
Comment (clintoncwolfe) [reply lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/README.md]: I've done so on #3491

**Review by clintoncwolfe — CHANGES_REQUESTED** [review]
I request that the template be moved under templates/profiles/gcp ; and that this PR be rebased off of cw/init-multi-platform

File: lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/attributes.yml (hunk 1 above)
Comment (skpaterson) [reply lib/plugins/inspec-init/lib/inspec-init/templates/profile_gcp/attributes.yml]: Hi @jerryaldrichiii, this was deliberately left to make life easier for the consumer, I've added a line to better reflect this in the latest update.

**Comment by skpaterson** [comment]
Hopefully this should address the PR feedback, let me know if we're good to proceed, thanks!

**Review by clintoncwolfe — DISMISSED** [review]

**Review by jerryaldrichiii — APPROVED** [review]

**Comment by clintoncwolfe** [comment]
Test failure on travis is a fluke, restarting test run.

UPDATE: it's a real failure, reproducible locally.  Rebase from master didn't fix, I'll look into it.  Added DNM for broken test

**Comment by clintoncwolfe** [comment]
I've rebased to (v3) master, and fixed the test: it was due to the test setup blindly appending to the YAML file.  I altered it to properly parse the YAML, add the dependency as a data structure, and then dump the YAML back out.

@jquick could you look this over; I'm recusing myself as an approver.

File: lib/plugins/inspec-init/lib/inspec-init/renderer.rb [hunk 4]
```diff
@@ -28,7 +28,10 @@ def render_with_values(template_type, template_values = {})
         template_values[:name] = template_values[:name].split(%r{\\|\/}).last
         # Generate the full full_destination_root_path path on disk
         full_destination_root_path = Pathname.new(Dir.pwd).join(profile_path)
-        ui.plain_text "Create new #{template_type} at #{ui.mark_text(full_destination_root_path)}"
+
+        # This is a bit gross
+        generator_type = template_subdir_path.split(%r{[\/]}).first.sub(/s$/, '')
```
Comment (jquick) [inline lib/plugins/inspec-init/lib/inspec-init/renderer.rb:33]: We may need to do a `Pathname.new(template_subdir_path).parent`(or something like that) here. I am worried about slashes on windows.

**Review by jquick — COMMENTED** [review]
This looks pretty good. Just one concern.

File: lib/plugins/inspec-init/lib/inspec-init/renderer.rb (hunk 4 above)
Comment (jquick) [reply lib/plugins/inspec-init/lib/inspec-init/renderer.rb:33]: I guess you do a File.join before this so it should covert any accordingly.

**Review by jquick — APPROVED** [review]

**Pull request merged**